backend/de.metas.business/src/main/java/de/metas/pricing/service/impl/PricingBL.java

```diff
@@ -279,7 +279,7 @@ private void setupPriceListAndDate(@NonNull final IEditablePricingContext pricin
 				&& pricingCtx.getPriceListId() == null
 				&& pricingCtx.getPriceListVersionId() == null)
 		{
-			final PricingSystemId pricingSystemId = bpartnerDAO.retrievePricingSystemIdOrNull(pricingCtx.getBPartnerId(), pricingCtx.getSoTrx());
+			final PricingSystemId pricingSystemId = bpartnerDAO.retrievePricingSystemIdOrNullInTrx(pricingCtx.getBPartnerId(), pricingCtx.getSoTrx());
```

Comment:
why this?

Reply:
The nre BPartner is not yet saved _outOfTrx_ so I get an error that no such partner exists for my ID. SO I am using the InTrx method to also covedthis case.

Reply:
![image](https://user-images.githubusercontent.com/16001442/112675345-b68add00-8e6f-11eb-9e96-399490ccb0de.png)

Reply:
add comment to it to shortly explain why are u doing this